File: src/problems/leetcode-cn/3315. 构造最小位运算数组 II/solution/python/solution.py
class Solution:
    def minBitwiseArray(self, nums: List[int]) -> List[int]:
        for i, e in enumerate(nums):
            if (e & 1 == 0):
                nums[i] = -1
            else:
                n = nums[i] + 1
                nums[i] ^= ((n & -n) >> 1)
        return nums

File: src/problems/leetcode-cn/3315. 构造最小位运算数组 II/solution/c++/solution.cpp
class Solution {
public:
    vector<int> minBitwiseArray(vector<int>& nums) {
        for_each(nums.begin(), nums.end(), [](int& i) {            
            if ((i & 1) == 0) i = -1;
            else {
                int n = i + 1;
                i ^= ((n & -n) >> 1);
            }
        });
        return nums;
    }
};